elements/pfe-modal/src/pfe-modal.scss

```diff
@@ -0,0 +1,142 @@
+@import "../../pfe-sass/pfe-sass";
+$LOCAL: modal;
+
+$pfe-modal--breakpoint--medium: 640px;
+
+:host {
+	--pfe-modal--MaxWidth: 70vw;
+	--pfe-modal--MinWidth: 50vw;
+	--pfe-modal--MaxWidth--mobile: 94vw;
+	--pfe-modal--MaxHeight: 90vh;
+	--pfe-modal--Padding: calc(#{pfe-var(container-padding)} * 2) 0 calc(#{pfe-var(container-padding)} * 2) calc(#{pfe-var(container-padding)} * 2);
+	--pfe-modal--Padding--mobile: 0 0 0 calc(#{pfe-var(container-padding)} * 2);
+	--pfe-modal__overlay--BackgroundColor: #{pfe-color(overlay)};
+	--pfe-modal--CloseButtonSize: calc(#{pfe-var(ui--element--size)} - 4px);
+	display: block;
+	position: relative;
+}
+
+:host(:not([has_header])) {
+	--pfe-modal--Padding: calc(#{pfe-var(container-padding)} * 2) calc(#{pfe-var(container-padding)} * 4) calc(#{pfe-var(container-padding)} * 2) calc(#{pfe-var(container-padding)} * 2);
+}
+
+.pfe-modal {
+	&__outer {
+		display: flex;
+		position: fixed;
+		height: 100%;
+		width: 100%;
+		top: 0;
+		left: 0;
+		flex-direction: column;
+		align-items: center;
+		justify-content: center;
+		z-index: var(--pfe-z--modal, 2000);
+
+		&[hidden] {
+			display: none;
+			z-index: 0;
+		}
+	}
+	&__overlay {
+		position: fixed;
+		height: 100%;
+		width: 100%;
+		top: 0;
+		left: 0;
+		background-color: pfe-local($cssvar: BackgroundColor, $region: overlay);
+		cursor: pointer;
+
+		&[hidden] {
+			display: none;
+		}
+	}
+	&__window {
+		position: relative;
+		max-width: pfe-local(MaxWidth--mobile);
+		min-width: pfe-local(MinWidth);
+		max-height: calc(100% + -32px);
+		margin: 0 auto;
+		box-shadow: pfe-var(box-shadow--lg);
+		background-color: pfe-color(surface--lightest);
+		border-radius: #{pfe-var(ui--border-radius)};
+
+		@media screen and (min-width: $pfe-modal--breakpoint--medium) {
+			max-width: pfe-local(MaxWidth);
+		}
+	}
+	&__container {
+		position: relative;
+		max-width: calc(100% - #{pfe-var(ui--element--size)});
+		max-height: inherit;
+		padding: var(--pfe-modal--Padding);
+
+		@media screen and (max-height: $pfe-modal--breakpoint--medium) {
+			padding: var(--pfe-modal--Padding--mobile);
+		}
+
+		&[hidden] {
+			display: none;
+		}
+
+		:host(:not([has_header])) & {
+			@media screen and (min-width: $pfe-modal--breakpoint--medium) {
+				padding: calc(#{pfe-var(container-padding)} * 2) calc(#{pfe-var(container-padding)} * 6) calc(#{pfe-var(container-padding)} * 2) calc(#{pfe-var(container-padding)} * 2);			
+			}
+
+			@media screen and (max-height: $pfe-modal--breakpoint--medium) and (min-width: $pfe-modal--breakpoint--medium) {
+				padding: 0 0 calc(#{pfe-var(container-padding)} * 2) calc(#{pfe-var(container-padding)} * 2);			
+			}
+
+			// Remove margin-top on the first slotted element that is not the header.
+			::slotted(*:nth-child(1)),
+			::slotted(*:nth-child(2)) {
+				margin-top: 0 !important;
+			}
+		}
+		::slotted([slot="pfe-modal--header"]) {
+			margin-top: 0 !important;
+			width: 90%;
```

Comment:
Should this be max-width?

Reply:
Would it be more accurate to do: `width: calc(100% - #{pfe-local($cssvar: size, $region: close)} - #{pfe-var(content-spacer)});`?  Basically subtract the size of the close button from the width of the header plus a little content-spacer breathing room between the header and the button?

Reply:
@castastrophe Rather than adjust the header width I set the `pfe-modal__content` region to get narrower so that the close button doesn't overlap. Testing this on the cdoherty sandbox but it seems to be working correctly.